ibis/backends/pandas/client.py

```diff
@@ -133,7 +168,19 @@ def infer_pandas_timestamp(value):
 
 @dt.infer.register(np.ndarray)
 def infer_array(value):
-    # TODO(kszucs): infer series
+    np_dtype_name = value.dtype.name
+    if np_dtype_name == 'object':
+        # The dtype of the np.array is ambiguous, so we need to infer the type
+        # of the elements manually.
+        if value.size == 0:
+            return dt.Array(dt.null)
+        return dt.Array(dt.highest_precedence(map(dt.infer, value)))
+    elif np_dtype_name == 'str32':
```

Comment:
do we have a good expl of why this is? this is a very specific numpy dtype. why isn't this just `np.str_`?

Reply:
It seems like the `dtype` of a `np.array` of strings is not `np.str_`. The dtype name here seems the most specific thing to check, but do you know of a better way?
```
>>> arr = np.array(['a', 'b', 'c'])
>>> arr.dtype
dtype('<U1')
>>> arr.dtype == np.str_
False
>>> arr.dtype.name
'str32'
```

Reply:
```
In [79]: arr.dtype.type                                                                                                                                  
Out[79]: numpy.str_

```

Reply:
ah perfect!

Reply:
Changed to check `.dtype.type ` now.